src/conn_sock.h

```diff
@@ -4,9 +4,33 @@
 #include <glib.h>   /* gboolean */
 #include "config.h" /* CONN_SOCK_BUF_SIZE */
 
+#define SOCK_TYPE_CONSOLE 1
+#define SOCK_TYPE_NOTIFY 2
+#define SOCK_IS_CONSOLE(sock_type) ((sock_type) == SOCK_TYPE_CONSOLE)
+#define SOCK_IS_NOTIFY(sock_type) ((sock_type) == SOCK_TYPE_NOTIFY)
+#define SOCK_IS_STREAM(sock_type) ((sock_type) == SOCK_TYPE_CONSOLE)
+#define SOCK_IS_DGRAM(sock_type) ((sock_type) != SOCK_TYPE_CONSOLE)
+
 /* Used for attach */
-struct conn_sock_s {
+/* I'm not sure on the nomenclature here.
+   in_sock and out_sock doesn't seem right, because ctr_stdio
+   breaks encapsulation and writes directly to the console
+   sockets.
+   In most cases in conn_sock.c, this struct is "INPUT" and
+   the next one is "OUTPUT".  Really it's this struct is "one"
+   and the next is "many", but I don't want the same fd in
+   two different sockets.
+
+   rmt seems to indicate "Remote User" i.e. attached console, or
+       "container's /dev/log" or "container's /run/notify"
+   lcl seems to incidate "A socket we own, locally" i.e. masterfd_stdin
+      or "host /dev/log" or "host /run/systemd/notify"
+  */
```

Comment:
I think you've chosen the right nomenclature here, can you update the comment to be more sure

Reply:
Heh sure